hudi-spark-datasource/hudi-spark/src/main/scala/org/apache/hudi/HoodieFileIndex.scala

```diff
@@ -125,6 +123,25 @@ case class HoodieFileIndex(
     }
   }
 
+  /**
+   * This method traverses StructType recursively to build map of columnName -> StructField
+   * Note : If there is nesting of columns like ["a.b.c.d", "a.b.c.e"]  -> final map will have keys corresponding
+   * only to ["a.b.c.d", "a.b.c.e"] and not for subsets like ["a.b.c", "a.b"]
+   * @param structField
+   * @return map of ( columns names -> StructField )
+   */
+  private def convertStructToMap(structField : Either[StructField,StructType]) : Map[String,StructField] = {
```

Comment:
@xiarixiaoyao : can you review this patch please.

Reply:
nit: can we name this method "generateNameFieldMap" or "fetchNameFieldMap"

Reply:
yes， will do it

Reply:
> generateNameFieldMap

Done